# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../openoffice-org/openoffice.org-2.4.0.ucb.newneon.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

http://www.openoffice.org/issues/show_bug.cgi?id=86123

diff -ru OOH680_m5/ucb.orig/source/ucp/webdav/LinkSequence.cxx OOH680_m5/ucb/source/ucp/webdav/LinkSequence.cxx
--- OOH680_m5/ucb.orig/source/ucp/webdav/LinkSequence.cxx	2008-02-12 20:13:08.000000000 +0000
+++ OOH680_m5/ucb/source/ucp/webdav/LinkSequence.cxx	2008-02-12 20:14:17.000000000 +0000
@@ -180,7 +180,7 @@
                       rInData.getStr() + nStart,
                       nEnd - nStart + TOKEN_LENGTH );
 
-#if NEON_VERSION >= 0250
+#if NEON_VERSION >= 0x0250
         success = !ne_xml_failed( parser );
 #else
         success = !!ne_xml_valid( parser );
diff -ru OOH680_m5/ucb.orig/source/ucp/webdav/LockEntrySequence.cxx OOH680_m5/ucb/source/ucp/webdav/LockEntrySequence.cxx
--- OOH680_m5/ucb.orig/source/ucp/webdav/LockEntrySequence.cxx	2008-02-12 20:13:08.000000000 +0000
+++ OOH680_m5/ucb/source/ucp/webdav/LockEntrySequence.cxx	2008-02-12 20:14:23.000000000 +0000
@@ -205,7 +205,7 @@
                        rInData.getStr() + nStart,
                        nEnd - nStart + TOKEN_LENGTH );
 
-#if NEON_VERSION >= 0250
+#if NEON_VERSION >= 0x0250
         success = !ne_xml_failed( parser );
 #else
         success = !!ne_xml_valid( parser );
diff -ru OOH680_m5/ucb.orig/source/ucp/webdav/LockSequence.cxx OOH680_m5/ucb/source/ucp/webdav/LockSequence.cxx
--- OOH680_m5/ucb.orig/source/ucp/webdav/LockSequence.cxx	2008-02-12 20:13:08.000000000 +0000
+++ OOH680_m5/ucb/source/ucp/webdav/LockSequence.cxx	2008-02-12 20:14:28.000000000 +0000
@@ -338,7 +338,7 @@
                       rInData.getStr() + nStart,
                       nEnd - nStart + TOKEN_LENGTH );
 
-#if NEON_VERSION >= 0250
+#if NEON_VERSION >= 0x0250
         success = !ne_xml_failed( parser );
 #else
         success = !!ne_xml_valid( parser );
diff -ru OOH680_m5/ucb.orig/source/ucp/webdav/makefile.mk OOH680_m5/ucb/source/ucp/webdav/makefile.mk
--- OOH680_m5/ucb.orig/source/ucp/webdav/makefile.mk	2008-02-12 20:13:08.000000000 +0000
+++ OOH680_m5/ucb/source/ucp/webdav/makefile.mk	2008-02-12 20:14:32.000000000 +0000
@@ -68,7 +68,7 @@
 .INCLUDE: $(SOLARINCDIR)$/$(NEONINCDIR)$/version.mk
 .ENDIF
 
-CFLAGS+= -DNEON_VERSION=$(NEON_VERSION)
+CFLAGS+= -DNEON_VERSION=0x$(NEON_VERSION)
 
 .IF "$(SYSTEM_NEON)" == "YES"
 CFLAGS+= $(NEON_CFLAGS)
diff -ru OOH680_m5/ucb.orig/source/ucp/webdav/NeonHeadRequest.cxx OOH680_m5/ucb/source/ucp/webdav/NeonHeadRequest.cxx
--- OOH680_m5/ucb.orig/source/ucp/webdav/NeonHeadRequest.cxx	2008-02-12 20:13:08.000000000 +0000
+++ OOH680_m5/ucb/source/ucp/webdav/NeonHeadRequest.cxx	2008-02-12 20:14:40.000000000 +0000
@@ -56,7 +56,7 @@
 
 namespace {
 
-#if NEON_VERSION >= 0250
+#if NEON_VERSION >= 0x0250
 void process_headers(ne_request *req,
                      DAVResource &rResource,
                      const std::vector< ::rtl::OUString > &rHeaderNames)
@@ -193,14 +193,14 @@
                                             inPath,
                                             RTL_TEXTENCODING_UTF8 ) );
 
-#if NEON_VERSION < 0250
+#if NEON_VERSION < 0x0250
     NeonHeadRequestContext aCtx( &ioResource, &inHeaderNames );
     ne_add_response_header_catcher( req, NHR_ResponseHeaderCatcher, &aCtx );
 #endif
 
     nError = ne_request_dispatch( req );
 
-#if NEON_VERSION >= 0250
+#if NEON_VERSION >= 0x0250
     process_headers(req, ioResource, inHeaderNames);
 #endif
 
diff -ru OOH680_m5/ucb.orig/source/ucp/webdav/NeonPropFindRequest.cxx OOH680_m5/ucb/source/ucp/webdav/NeonPropFindRequest.cxx
--- OOH680_m5/ucb.orig/source/ucp/webdav/NeonPropFindRequest.cxx	2008-02-12 20:13:08.000000000 +0000
+++ OOH680_m5/ucb/source/ucp/webdav/NeonPropFindRequest.cxx	2008-02-12 20:14:54.000000000 +0000
@@ -176,7 +176,7 @@
 
 // -------------------------------------------------------------------
 extern "C" void NPFR_propfind_results( void* userdata,
-#if NEON_VERSION >= 0260
+#if NEON_VERSION >= 0x0260
                                        const ne_uri* uri,
 #else
                                        const char* href,
@@ -185,7 +185,7 @@
 {
 	// @@@ href is not the uri! DAVResource ctor wants uri!
 
-#if NEON_VERSION >= 0260
+#if NEON_VERSION >= 0x0260
 	DAVResource theResource(
 						OStringToOUString( uri->path, RTL_TEXTENCODING_UTF8 ) );
 #else
@@ -218,7 +218,7 @@
 
 // -------------------------------------------------------------------
 extern "C" void NPFR_propnames_results( void* userdata,
-#if NEON_VERSION >= 0260
+#if NEON_VERSION >= 0x0260
                                         const ne_uri* uri,
 #else
                                         const char* href,
@@ -227,7 +227,7 @@
 {
 	// @@@ href is not the uri! DAVResourceInfo ctor wants uri!
 	// Create entry for the resource.
-#if NEON_VERSION >= 0260
+#if NEON_VERSION >= 0x0260
 	DAVResourceInfo theResource(
 						OStringToOUString( uri->path, RTL_TEXTENCODING_UTF8 ) );
 #else
diff -ru OOH680_m5/ucb.orig/source/ucp/webdav/NeonSession.cxx OOH680_m5/ucb/source/ucp/webdav/NeonSession.cxx
--- OOH680_m5/ucb.orig/source/ucp/webdav/NeonSession.cxx	2008-02-12 20:13:08.000000000 +0000
+++ OOH680_m5/ucb/source/ucp/webdav/NeonSession.cxx	2008-02-12 20:15:08.000000000 +0000
@@ -252,7 +252,7 @@
 // A simple Neon response_block_reader for use with an XInputStream
 // -------------------------------------------------------------------
 
-#if NEON_VERSION >= 0250
+#if NEON_VERSION >= 0x0250
 extern "C" int NeonSession_ResponseBlockReader(void * inUserData, 
 #else
 extern "C" void NeonSession_ResponseBlockReader(void * inUserData, 
@@ -272,7 +272,7 @@
         if ( xInputStream.is() )
             xInputStream->AddToStream( inBuf, inLen );
     }
-#if NEON_VERSION >= 0250
+#if NEON_VERSION >= 0x0250
     return 0;
 #endif
 }
@@ -282,7 +282,7 @@
 // A simple Neon response_block_reader for use with an XOutputStream
 // -------------------------------------------------------------------
 
-#if NEON_VERSION >= 0250
+#if NEON_VERSION >= 0x0250
 extern "C" int NeonSession_ResponseBlockWriter( void * inUserData, 
 #else
 extern "C" void NeonSession_ResponseBlockWriter( void * inUserData, 
@@ -304,7 +304,7 @@
             xOutputStream->writeBytes( aSeq );
         }
     }
-#if NEON_VERSION >= 0250
+#if NEON_VERSION >= 0x0250
     return 0;
 #endif
 }
@@ -1667,7 +1667,7 @@
     void *cursor = NULL;
     const char *name, *value;
 
-#if NEON_VERSION < 0250
+#if NEON_VERSION < 0x0250
     if ( getheaders )
 	ne_add_response_header_catcher( req, runResponseHeaderHandler, userdata );
 #endif
@@ -1675,7 +1675,7 @@
 
     ret = ne_request_dispatch( req );
 
-#if NEON_VERSION >= 0250
+#if NEON_VERSION >= 0x0250
     if ( getheaders )
     {
         while ((cursor = ne_response_header_iterate(req, cursor, &name, &value)) 
diff -ru OOH680_m5/ucb.orig/source/ucp/webdav/NeonUri.cxx OOH680_m5/ucb/source/ucp/webdav/NeonUri.cxx
--- OOH680_m5/ucb.orig/source/ucp/webdav/NeonUri.cxx	2008-02-12 20:13:08.000000000 +0000
+++ OOH680_m5/ucb/source/ucp/webdav/NeonUri.cxx	2008-02-12 20:15:27.000000000 +0000
@@ -73,34 +73,34 @@
 namespace {
 
 const ne_uri g_sUriDefaultsHTTP  = { "http",  
-#if NEON_VERSION >= 0260
+#if NEON_VERSION >= 0x0260
                                      NULL, 
 #endif
                                      NULL, 
                                      DEFAULT_HTTP_PORT, 
-#if NEON_VERSION >= 0260
+#if NEON_VERSION >= 0x0260
                                      NULL, 
 #endif
                                      NULL, 
                                      NULL };
 const ne_uri g_sUriDefaultsHTTPS = { "https", 
-#if NEON_VERSION >= 0260
+#if NEON_VERSION >= 0x0260
                                      NULL, 
 #endif
                                      NULL, 
                                      DEFAULT_HTTPS_PORT, 
-#if NEON_VERSION >= 0260
+#if NEON_VERSION >= 0x0260
                                      NULL, 
 #endif
                                      NULL, 
                                      NULL };
 const ne_uri g_sUriDefaultsFTP   = { "ftp",   
-#if NEON_VERSION >= 0260
+#if NEON_VERSION >= 0x0260
                                      NULL, 
 #endif
                                      NULL, 
                                      DEFAULT_FTP_PORT, 
-#if NEON_VERSION >= 0260
+#if NEON_VERSION >= 0x0260
                                      NULL, 
 #endif
                                      NULL, 
@@ -188,7 +188,7 @@
                     pUri->scheme ? pUri->scheme : pUriDefs->scheme,
                     RTL_TEXTENCODING_UTF8 );
     mUserInfo = rtl::OStringToOUString(
-#if NEON_VERSION >= 0260
+#if NEON_VERSION >= 0x0260
                     pUri->userinfo ? pUri->userinfo : pUriDefs->userinfo,
 #else
                     pUri->authinfo ? pUri->authinfo : pUriDefs->authinfo,
@@ -202,7 +202,7 @@
                     pUri->path ? pUri->path : pUriDefs->path,
                     RTL_TEXTENCODING_UTF8 );
 
-#if NEON_VERSION >= 0260
+#if NEON_VERSION >= 0x0260
     if ( pUri->query ) 
     {
         mPath += rtl::OUString::createFromAscii( "?" );
diff -ru OOH680_m5/ucb.orig/source/ucp/webdav/UCBDeadPropertyValue.cxx OOH680_m5/ucb/source/ucp/webdav/UCBDeadPropertyValue.cxx
--- OOH680_m5/ucb.orig/source/ucp/webdav/UCBDeadPropertyValue.cxx	2008-02-12 20:13:08.000000000 +0000
+++ OOH680_m5/ucb/source/ucp/webdav/UCBDeadPropertyValue.cxx	2008-02-12 20:15:32.000000000 +0000
@@ -393,7 +393,7 @@
 
         ne_xml_parse( parser, rInData.getStr(), rInData.getLength() );
 
-#if NEON_VERSION >= 0250
+#if NEON_VERSION >= 0x0250
         success = !ne_xml_failed( parser );
 #else
         success = !!ne_xml_valid( parser );
